src/createClient.js

```diff
@@ -17,28 +19,54 @@ function createClient (options) {
       client.init()
     } else {
       ping(client.options).then(ad => {
-        const adVersion = ad.version?.split('.').slice(0, 3).join('.') // Only 3 version units
-        client.options.version = options.version ?? (Options.Versions[adVersion] ? adVersion : Options.CURRENT_VERSION)
+        if (client.options.transport === 'raknet') {
+          const adVersion = ad.version?.split('.').slice(0, 3).join('.') // Only 3 version units
+          client.options.version = options.version ?? (Options.Versions[adVersion] ? adVersion : Options.CURRENT_VERSION)
 
-        if (ad.portV4 && client.options.followPort) {
-          client.options.port = ad.portV4
+          if (ad.portV4 && client.options.followPort) {
+            client.options.port = ad.portV4
+          }
+
+          client.conLog?.(`Connecting to ${client.options.host}:${client.options.port} ${ad.motd} (${ad.levelName}), version ${ad.version} ${client.options.version !== ad.version ? ` (as ${client.options.version})` : ''}`)
+        } else if (client.options.transport === 'nethernet') {
+          client.conLog?.(`Connecting to ${client.options.networkId} ${ad.motd} (${ad.levelName})`)
```

Comment:
How do we know what version to connect as here if the user does not specify/can't get via ping?

Reply:
OK, what is `advertisement.version` then? We may have to figure out the version after connecting to the server then like nmp can do. It should be sent in login/network_settings packet otherwise. The current code should also be matching version by protocol version num over strings but that should be fixed outside this PR

Reply:
Not entirely sure but I assume it's the version of NetherNet protocol being used as it's just a uint8. From my testing it's always been set to 3 over multiple version updates.